### PR TITLE
fix: debounce kubernetes listeners channels individually

### LINKED ISSUE #6277: Some kubernetes contexts changes are not dispatched
### Bug description

The ContextsManager is using some timer in `dispatch` to limit sending data too often. But because we pass options to `dispatch`, to optionally dispatch selected data, we can miss sending data if a more restrictive option overrides other options.

For example:
- time t: send all
- time t + 10ms: send services

At time t, sending all data will be delayed, and will be canceled by the new call at t+10ms (due to the short delay), but only services data will be sent, and other data will be lost.
 

### Operating system

all

### Installation Method

None

### Version

1.7.0

### Steps to reproduce

_No response_

### Relevant log output

_No response_

### Additional context

_No response_

### PR DESCRIPTION
**Pull request by feloy**
Signed-off-by: Philippe Martin <phmartin@redhat.com>

### What does this PR do?

Fixes the ContextsManager to debounce data sent individually

### What issues does this PR fix or reference?

Fix #6277

### How to test this PR?

Switch contexts, and check that all resources (pods, deployments, services, ingresses/routes) are updated every time

- [x] Tests are covering the bug fix or the new feature

### CODE REVIEW

**Review by deboer-tim — APPROVED** [review]
LGTM, no problems testing.

**Comment by deboer-tim** [comment]
MacOS check failed, restarted.

File: packages/main/src/plugin/kubernetes-context-state.ts [hunk 1]
```diff
@@ -772,48 +772,59 @@ export class ContextsManager {
...
-      this.states.getContextResources(this.kubeConfig.currentContext, resourceName),
+    clearTimeout(this.dispatchCurrentContextResourceTimers.get(resourceName));
+    this.dispatchCurrentContextResourceTimers.set(
+      resourceName,
+      setTimeout(() => {
+        this.apiSender.send(
+          `kubernetes-current-context-${resourceName}-update`,
+          this.states.getContextResources(this.kubeConfig.currentContext, resourceName),
+        );
+      }, dispatchTimeout),
```
Comment (benoitf) [inline packages/main/src/plugin/kubernetes-context-state.ts]: this method looks like a copy /paste for each dispatch, should it be unified  with a method doing clear and dispatch ?

**Review by benoitf — COMMENTED** [review]
Hello, I'm not sure to get the latest refactoring change

why debounce has not a parameter like eventName, value, timer

debounce('kubernetes-contexts-general-state-update', this.states.getContextResources(this.kubeConfig.currentContext),
this.dispatchContextsGeneralStateTimer
?

**Review by benoitf — APPROVED** [review]
thanks

**Pull request merged**